src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp

```diff
@@ -149,41 +233,75 @@ struct InitializeDampedHarmonic {
-          "transformed to the inertial frame at each time step. Transforming "
-          "the spacetime derivative requires spacetime Hessians, which are not "
-          "implemented for the maps and there is currently no plan to add them "
-          "because we do not need them for anything else.");
+    if constexpr (UseRollon) {
+      if (not db::get<domain::CoordinateMaps::Tags::CoordinateMap<
+                  Metavariables::volume_dim, Frame::Grid, Frame::Inertial>>(box)
+                  .is_identity()) {
+        ERROR(
+            "Cannot use the damped harmonic rollon gauge with a moving mesh "
+            "because the rollon is not implemented for a moving mesh. The "
+            "issue is that the initial H_a needs to be in the grid frame, and "
+            "then transformed to the inertial frame at each time step. "
+            "Transforming the spacetime derivative requires spacetime "
+            "Hessians, which are not implemented for the maps and there is "
+            "currently no plan to add them because we do not need them for "
+            "anything else.");
+      }
+      const auto& inverse_jacobian =
+          db::get<domain::Tags::InverseJacobian<Dim, Frame::Logical, frame>>(
+              box);
+
+      auto [initial_gauge_h, initial_d4_gauge_h] = impl_rollon(
+          db::get<gr::Tags::SpacetimeMetric<Dim, Frame::Inertial, DataVector>>(
+              box),
+          db::get<GeneralizedHarmonic::Tags::Pi<Dim, Frame::Inertial>>(box),
+          db::get<GeneralizedHarmonic::Tags::Phi<Dim, Frame::Inertial>>(box),
+          db::get<domain::Tags::Mesh<Dim>>(box), inverse_jacobian);
+
+      // Add gauge tags
+      using compute_tags =
+          db::AddComputeTags<DampedHarmonicRollonCompute<frame>>;
+
+      // Finally, insert gauge related quantities to the box
+      return std::make_tuple(
+          Initialization::merge_into_databox<
+              InitializeDampedHarmonic,
+              db::AddSimpleTags<
+                  GeneralizedHarmonic::Tags::InitialGaugeH<Dim, frame>,
+                  GeneralizedHarmonic::Tags::SpacetimeDerivInitialGaugeH<
+                      Dim, frame>>,
+              compute_tags>(std::move(box), std::move(initial_gauge_h),
+                            std::move(initial_d4_gauge_h)));
+    } else {
+      const double initial_time =
+          db::get<::Initialization::Tags::InitialTime>(box);
+      const auto inertial_coords =
+          db::get<::domain::CoordinateMaps::Tags::CoordinateMap<
+              Metavariables::volume_dim, Frame::Grid, Frame::Inertial>>(box)(
+              db::get<::domain::Tags::ElementMap<Metavariables::volume_dim,
+                                                 Frame::Grid>>(box)(
+                  db::get<domain::Tags::Coordinates<Dim, Frame::Logical>>(box)),
+              initial_time, db::get<::domain::Tags::FunctionsOfTime>(box));
+
+      auto gauge_corrected_pi = new_pi_from_gauge_h(
```

Comment:
is it not better to make the `new_pi_from_gauge_h` function a return-by-pointer function and use it as the mutator for the `db::mutate` call?

Reply:
I personally dislike the "mutator" business, finding it rather difficult to read.

Reply:
I'm fine with leaving a lambda in the db::mutate call, but I do think it's better to perform the calculation in a return-by-pointer function; I think you can save an allocation/copy, right?